torch/nn/modules/_functions.py

```diff
@@ -7,7 +7,11 @@ class SyncBatchNorm(Function):
     @staticmethod
     def forward(self, input, weight, bias, running_mean, running_var, eps, momentum, process_group, world_size):
         input = input.contiguous()
-        count = torch.Tensor([input.numel() // input.size(1)]).to(input.device)
+
+        size = input.numel() // input.size(1)
+        if size == 1:
+            raise ValueError('Expected more than 1 value per channel when training, got input size {}'.format(size))
```

Comment:
Isn't this method called both for training and testing?

Reply:
Good point, missed the training flag: https://github.com/pytorch/pytorch/blob/5a8ad66354c81f1fbe1e68848b420cad7a2ce97b/torch/nn/functional.py#L1651-L1666

Reply:
Forgot how to English.

The default behavior is to only raise exception for `self.training or not self.track_running_stats`, which is the same logic that would take you to `SyncBatchNorm`.
So we should be good here.

Reply:
The missing link here: https://github.com/pytorch/pytorch/blob/5a8ad66354c81f1fbe1e68848b420cad7a2ce97b/torch/nn/modules/batchnorm.py#L78

Reply:
Thanks for the links !